llvm/lib/IR/Type.cpp

```diff
@@ -214,6 +220,18 @@ int Type::getFPMantissaWidth() const {
   return -1;
 }
 
+int Type::getDFPPrecisionInDigits() const {
+  if (auto *VTy = dyn_cast<VectorType>(this))
+    return VTy->getElementType()->getDFPPrecisionInDigits();
+  assert(isDecimalFloatingPointTy() && "Not a decimal floating point type!");
+  // Precision values following the table in section X.2.1 of WG14 N2601.
+  if (getTypeID() == Decimal32TyID) return 7;
+  if (getTypeID() == Decimal64TyID) return 16;
+  if (getTypeID() == Decimal128TyID) return 34;
+  assert("unknown decimal floating point type");
```

Comment:
What's the idiom here?  Isn't the condition in the assert always logical true?

Reply:
```suggestion
  report_fatal_error("unknown decimal floating point type");
```
Good catch! I recommend using `report_fatal_error()` here (include `llvm/Support/ErrorHandling.h`).

Reply:
In which case, since `report_fatal_error()` is declared `[[noreturn]]`, the following return statement can be removed as well.